include/deal.II/grid/grid_tools.h

```diff
@@ -556,6 +556,139 @@ namespace GridTools
   get_active_neighbors (const typename Container::active_cell_iterator        &cell,
                         std::vector<typename Container::active_cell_iterator> &active_neighbors);
 
+  /**
+   * Extract and return the active cell layer around a subdomain (set of active
+   * cells) in the @p container (i.e. those that share a common set of vertices
+   * with the subdomain but are not a part of it).
+   * Here, the "subdomain" consists of exactly all of those cells for which the
+   * @p predicate returns @p true. Examples of predicates that are frequently
+   * useful can be found in namespace IteratorFilters.
+   *
+   * @note This function may return both locally owned and ghost cells, but
+   * is expected not to return artificial cells.
```

Comment:
Ok, I think that what I'll do to make this all a lot more clear is remove the assert that checks for artificial cells from the primary function and selectively include it in all of the individual functions. Then its clear as to what type of cell each function is and isn't allowed to return.